statistics/handle/handle.go

```diff
@@ -318,9 +318,9 @@ func (h *Handle) MergePartitionStats2GlobalStats(sc sessionctx.Context, opts map
 		return
 	}
 	globalTableInfo := globalTable.Meta()
-	partitionNum := globalTableInfo.Partition.Num
```

Comment:
The `globalTableInfo.Partition.Num` do not record the added partition. So we should use the `len(globalTableInfo.Partition.Definitions)` to replace it.